front/lib/iam/invitations.ts

```diff
@@ -68,6 +74,39 @@ export async function getPendingMembershipInvitationForEmailAndWorkspace(
   });
 }
 
+export async function getPendingMembershipInvitationWithWorkspaceForEmail(
+  email: string
+): Promise<{
+  invitation: MembershipInvitationType;
+  workspace: LightWorkspaceType;
+} | null> {
+  const pendingInvitation = await MembershipInvitation.findOne({
+    where: {
+      inviteEmail: email,
+      status: "pending",
+    },
+    include: [Workspace],
+  });
```

Comment:
What if the pending invitation is not valid anymore ? User will be redirected to a page with expired token ?

Reply:
Seen IRL, worst case they get's redirected to the join and the invitation has expired (expiration is on the token). They will eventually get redirected to an error page, but they second sign up will lead them to a new workspace.